src/journal-remote/journal-upload.c

```diff
@@ -189,10 +193,19 @@ int start_upload(Uploader *u,
                 _cleanup_(curl_slist_free_allp) struct curl_slist *h = NULL;
                 struct curl_slist *l;
 
-                h = curl_slist_append(NULL, "Content-Type: application/vnd.fdo.journal");
+                _cleanup_free_ char *header = strjoin("Content-Encoding: ", compression_lowercase_to_string(u->compression.algorithm));
```

Comment:
The compression algorithm may be NONE. Please set the encoding only when really compressed.

BTW, please move the header entry at the last. Adding at first complicates the code and commit diff.